src/vs/workbench/api/common/extHostTypeConverters.ts

```diff
@@ -841,17 +844,27 @@ export namespace CompletionItem {
 		result.filterText = suggestion.filterText;
 		result.preselect = suggestion.preselect;
 		result.commitCharacters = suggestion.commitCharacters;
-		result.range = editorRange.Range.isIRange(suggestion.range) ? Range.to(suggestion.range) : undefined;
+		result.range = convertRange(suggestion.range);
 		result.range2 = editorRange.Range.isIRange(suggestion.range) ? undefined : { inserting: Range.to(suggestion.range.insert), replacing: Range.to(suggestion.range.replace) };
 		result.keepWhitespace = typeof suggestion.insertTextRules === 'undefined' ? false : Boolean(suggestion.insertTextRules & modes.CompletionItemInsertTextRule.KeepWhitespace);
-		// 'inserText'-logic
+		// 'insertText'-logic
 		if (typeof suggestion.insertTextRules !== 'undefined' && suggestion.insertTextRules & modes.CompletionItemInsertTextRule.InsertAsSnippet) {
 			result.insertText = new types.SnippetString(suggestion.insertText);
 		} else {
 			result.insertText = suggestion.insertText;
 			result.textEdit = result.range instanceof types.Range ? new types.TextEdit(result.range, result.insertText) : undefined;
 		}
-		// TODO additionalEdits, command
+		if (suggestion.additionalTextEdits) {
+			const validAdditionalEdits = [];
+			for (const additionalEdit of suggestion.additionalTextEdits) {
+				const range = convertRange(additionalEdit.range);
+				if (range instanceof types.Range) {
+					validAdditionalEdits.push(new types.TextEdit(range, additionalEdit.text));
+				}
+			}
+			result.additionalTextEdits = validAdditionalEdits.length ? validAdditionalEdits : undefined;
+		}
+		// TODO command
```

Comment:
For commands, use https://github.com/microsoft/vscode/blob/ca2acdc8b67c45a4a45d316b06d3ae523a28e0b1/src/vs/workbench/api/common/extHostCommands.ts#L267 (which you somehow need to get here)